### DISCUSSION TITLE
Issue with cache.entrySet(filter,compartor)

### DISCUSSION THREAD

**Discussion by Manikanta532** [oracle/coherence#115] [Q&A]
I have an issue, where result set retrieved from cache is empty on pageNumber >0 using cache.entrySet(filter,compartor).
I am getting result using the method when pageNumber is 0. What could be the reason?. It gives the records when null / default comparator is passed. It always retrieves empty record on pageNumber > 0 .

My custom comparator is simple one on String using s1.compareTo(s2)

**Comment by mgamanho** [comment]
Hi, are you using a limit filter? If so you need to make sure you make the call from the same context (thread/method) as the filter internally keeps track of things like top/bottom anchor and cookie so relative pages can be correlated.

You can also try with the default comparator which should work in most cases, before you start debugging yours.

If your use-case is different please try and post as many details as possible here.

**Reply by Manikanta532** [reply]
Hi @mgamanho
Yes, using limit filter. we will fire request each time from web service when we needs next page data. Looks like that is the issue here thread differs in each request. In our case default comparator is behaving inconsistently for different caches. we are getting contract issues.

**Comment by mgamanho** [comment]
Yes you need to find a way to keep a reference to the filter around and use the same for different pages.
As stated in the javadoc: `This filter is a mutable object that is modified by the query processor. Clients are supposed to hold a reference to this filter and repetitively pass it to query methods after setting a desired page context calling setPage(int), nextPage(), or previousPage().`
Let us know if you have further questions.

**Reply by Manikanta532** [reply]
I tried maintaining the same thread for test purpose and increase the page number as suggested using next page and passed comparator with entryset(limitfilter,comparator). It still does not send the data for second iteration with pageNo 1.

set = cache.entrySet(limitFilter, comparator)--Data Retrieved where pageNo=0
limitFilter.nextPage()
set = cache.entrySet(limitFilter, comparator)--Data is not coming where pageNo =1.

There is data always in the cache , I am passing pagesize as 10.

My comparator logic is am getting the key of the records in the cache.
using String.compareTo(String).

Any clue here, why the behavior is not as expected.
If possible, could you please share the examples for it.  Am new to coherence and trying to fix a bug which is failing for pagination.

Is there any alternative for limit filter which take similarly pageno and page size.

**Reply by mgamanho** [reply]
Did you try without a comparator? We'll use the default for the type.
Yours is otherwise the correct way.
Can you post how you create the filter? Also, I assume your keys are of type String?
You can find a bunch of examples in the tests section (prj/test/functional/filter/LimitFilterTests.java I believe).
Thanks

**Reply by Manikanta532** [reply]
Could you please share the example section link  . We use the coherence jar 14.1.x I don't see any samples in it.

**Reply by chpatel3** [reply]
@Manikanta532, As @mgamanho pointed earlier you can find the examples under the `prj/test` dir of the project itself i.e. https://github.com/oracle/coherence/blob/v14.1.1.0/prj/test/functional/filter/src/test/java/filter/LimitFilterTests.java
